solo/models.py

```diff
@@ -1,63 +1,65 @@
+from __future__ import annotations
+
+from typing import Any, TypeVar
+
 from django.conf import settings
+from django.core.cache import caches
 from django.db import models
 
-try:
-    from django.core.cache import caches
-    get_cache = lambda cache_name: caches[cache_name]
-except ImportError:
-    from django.core.cache import get_cache
-
 from solo import settings as solo_settings
 
 DEFAULT_SINGLETON_INSTANCE_ID = 1
 
+Self = TypeVar("Self", bound="SingletonModel")
```

Comment:
I'm not 100% clear how this is being used, but could we instead use the official `Self` type that is included in Python 3.11 and backported to older versions using [`typing-extenstions`](https://typing-extensions.readthedocs.io/en/latest/#Self)?

This would add `typing-extentions` as a dependency to the project, but it is a very common dependency. We could add a `TODO` somewhere that says to remove the dependency after dropping support for > 3.11.

Reply:
The current solution I provided was the way of doing this before `Self`. `typing.Self` is a convenience feature, but indeed does add an extra dependency.

Quoting my OP:

> `typing.TypeVar` is used as a workaround until `typing.Self` can be used without making use of `typing_extensions`.

So if it's fine to have `typing-extensions`, I'll go for it!

Reply:
> We could add a `TODO` somewhere that says to remove the dependency after dropping support for > 3.11.

The added Python version check is in my opinion explicit enough. If you want me to add an extra TODO, let me know!